packages/runtime/container-runtime/src/test/runningSummarizer.spec.ts

```diff
@@ -128,58 +127,59 @@ describe("Runtime", () => {
-                                    treeNodeCount: 0,
-                                    blobNodeCount: 0,
-                                    handleNodeCount: 0,
-                                    totalBlobSize: 0,
-                                    dataStoreCount: 0,
-                                    summarizedDataStoreCount: 0,
-                                    unreferencedBlobSize: 0,
-                                },
-                                handle: "test-handle",
-                                clientSequenceNumber: lastClientSeq,
-                            } as const;
-                        },
-                        stop(reason?: SummarizerStopReason) {
-                            stopCall++;
-                        },
+                    // submitSummaryCallback
+                    async (options) => {
+                        runCount++;
+
+                        const { fullTree = false, refreshLatestAck = false } = options;
+                        if (fullTree) {
+                            fullTreeRunCount++;
+                        }
+                        if (refreshLatestAck) {
+                            refreshLatestAckRunCount++;
+                        }
+
+                        // immediate broadcast
+                        emitBroadcast();
+
+                        if (shouldDeferGenerateSummary) {
+                            deferGenerateSummary = new Deferred<void>();
+                            await deferGenerateSummary.promise;
+                            deferGenerateSummary = undefined;
+                        }
+                        return {
+                            stage: "submit",
+                            referenceSequenceNumber: lastRefSeq,
+                            generateDuration: 0,
+                            uploadDuration: 0,
+                            submitOpDuration: 0,
+                            summaryTree: { type: SummaryType.Tree, tree: {} },
+                            summaryStats: {
+                                treeNodeCount: 0,
+                                blobNodeCount: 0,
+                                handleNodeCount: 0,
+                                totalBlobSize: 0,
+                                dataStoreCount: 0,
+                                summarizedDataStoreCount: 0,
+                                unreferencedBlobSize: 0,
+                            },
+                            handle: "test-handle",
+                            clientSequenceNumber: lastClientSeq,
+                        } as const;
                     },
                     new SummarizeHeuristicData(0, { refSequenceNumber: 0, summaryTime: Date.now() }),
                     () => { },
                     summaryCollection,
+                    // ICancellable
+                    {
+                        cancelled: false,
+                        waitCancelled: new Promise(() => {}),
```

Comment:
isn't this a memory leak? this promise will never resolve or reject

Reply:
anyone who waits on it will wait forever

Reply:
As any other object, it would get collected when there are no references to it. It will only create forever leak if code using it never finishes (for other reasons), which would be a bug to investigate / fix.

Reply:
That's by design - this path is never cancelled. So if someone waits for cancelation, then it will never happen.
It's no different than let's say we added cancelation signal to all our fetches, and test code would pass signal that never gets activated - it's the right mechanism, just happened that this mechanism is never engaged in this (test) flow.

Maybe other way to say it - this flow does not test cancellation, thus it uses promise that never resolves or rejects.